### PR TITLE
Add an FAQ sections on different types of outputs

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+## Where are the generated files?
+
+There are 3 places you might see files getting generated. On _every_ build files
+might go to _cache_ (`.dart_tool/build/generated/*`), and _source_ (the working
+tree of your package). These are determined by the `build_to` configuration of
+the Builders you are using. When a Builder specifies `build_to: source` you will
+see them next to your other files but should not edit them by hand, and they are
+generally meant to be published with your package. If you see a warning about
+"conflicting outputs" they refer to _source_ outputs, because when the build
```
Comment (jakemac53) [inline docs/faq.md]: they _usually_ refer to _source_ outputs

File: docs/faq.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+tool is starting with no prior information it can't tell if it might be
+overwriting some file you wrote by hand.
+
+Separately the `--output` option can specify a directory to create a _merged_
```
Comment (jcollins-g) [inline docs/faq.md]: This is the part that was confusing to me.  I had assumed that specifying an --output directory meant that nothing would get written to my sources (instead, the built items would _only_ appear in my merged output directory).

On reflection, I can see why you would do something like this (analyzer/IDEs probably like this a lot better, for example).  But maybe something making it more clear that --output effectively copies _after_ writing into the source directory would have made me understand the behavior I was seeing was intentional.

File: docs/faq.md (hunk 1 above)
Comment (natebosch) [reply docs/faq.md]: When we say "conflicting" we mean source outputs, I think our other error is "duplicate" outputs for when 2 builders want to produce the same asset.

File: docs/faq.md (hunk 2 above)
Comment (natebosch) [reply docs/faq.md]: I'll update the `--help` output to call the output a "copy" - https://github.com/dart-lang/build/pull/1598

**Pull request merged**